src/pages/ContentScripts/features/repo-collection/Mydropdown.tsx

```diff
@@ -0,0 +1,309 @@
+      if (result.userCollectionData) {
+        console.log('hi,loading', result.userCollectionData);
+      }
+      setCollectionData(result.userCollectionData);
+      console.log('collectionData', collectionData);
+      const temp = getListData();
+      console.log('listData', temp);
+      setItems(temp);
+    });
+  }, [refresh]);
+
+  function handleValueClick(value: string) {
+    const temp = collectionData[value as keyof typeof collectionData];
+    setModalValue([value, temp]);
+    // open modal
+    setModalVisible(true);
+    setOpen(false);
+  }
+
+  const closeModal = () => {
+    // close modal
+    setModalVisible(false);
+    setOpen(true);
+  };
+
+  const getListData = () => {
+    type DataType = {
+      [key: string]: string[];
+    };
+    console.log('hello collection', collectionData);
+    const originalData: DataType = collectionData;
+    const transformedData: DataType = {};
+
+    // 遍历原始数据对象
+    for (const key in originalData) {
+      const values = originalData[key];
+
+      values.forEach((value) => {
+        if (!transformedData[value]) {
+          transformedData[value] = [];
+        }
+        transformedData[value].push(key);
+      });
+    }
+
+    console.log('trans', transformedData);
+
+    const labels: { key: string; label: JSX.Element }[] = [];
+
+    if (transformedData[repoName]) {
+      transformedData[repoName].forEach((value, index) => {
+        labels.push({
+          key: (index + 1).toString(),
+          label: <div onClick={() => handleValueClick(value)}>{value}</div>,
+        });
+      });
+    }
+
+    return labels;
+  };
+  const handleOpenChange = (newOpen: boolean) => {
+    setOpen(newOpen);
+  };
+
+  function addNewCollection() {
+    if (newCollectionName.trim() === '') {
+      alert('请输入集合名称');
+      return;
+    }
+    /*
+    const temp={...collectionData,newCollectionName:[""]};
+    console.log('newCN',temp);
+    setCollectionData(temp);
+    setItems(updateCollectionList);
+    chrome.storage.sync.set({ collectionData: collectionData }).then(() => {
+      console.log('UserCollection is set');
+    });
+ */
+  }
+  function backToList() {
+    // items变化
+    const temp = getListData();
+    setItems(temp);
+    // footer变化
+    setFooterContent(listFooter);
+  }
+
+  function applyChange() {
+    // items变化
+    const temp = getListData();
+    setItems(temp);
+    // footer变化
+    setFooterContent(listFooter);
+  }
+  const handleCheckboxChange = (e: CheckboxChangeEvent, key: string) => {
+    const newData = { ...collectionData }; // 创建数据的副本
+    console.log('newData', newData);
+    console.log('e', e);
+    if (e.target.checked) {
+      newData[key as keyof typeof newData].push(repoName);
+      console.log('add', newData);
+    } else {
+      newData[key as keyof typeof newData] = newData[
+        key as keyof typeof newData
+      ].filter((item) => item !== repoName);
+      console.log('remove', newData);
+    }
+
+    setCollectionData({
```

Comment:
The `handleCheckboxChange` function here is to handle the checked and unchecked situations. When checked, `newData` will increase the checked value, and when unchecked, `newData` will remove the checked value. In other words, `newData` can change smoothly, but the `setCollectionData` function only generates updates when adding data, and remains silent when reducing data.

Reply:
There seems to be a small deviation in locating the specific code location...

Reply:
> but the setCollectionData function only generates updates when adding data, and remains silent when reducing data.

I don't think so. Both conditions the `setCollectionData` works.

<img width="1120" alt="image" src="https://github.com/hypertrons/hypertrons-crx/assets/32434520/33cc4387-0b12-4e78-93f3-07f22d0dbc84">

See: https://react.dev/learn/updating-objects-in-state#treat-state-as-read-only

PS: `=` is a mutation operation.

In case you have questions on the output of the following line: `console.log('collectionData', collectionData);` you may also checkout [this](https://react.dev/learn/state-as-a-snapshot#state-over-time).

Reply:
I think [Checkbox Group](https://ant-design.antgroup.com/components/checkbox-cn#checkbox-group) can be applied here to get all statuses of the checkboxes at once. Intead to listen to each checkbox's onChange event individually which results in making things more complex.

<img width="498" alt="image" src="https://github.com/hypertrons/hypertrons-crx/assets/32434520/f14b883c-be73-4cda-be4e-6b314dcf0af2">

AND, don't store the <Checkbox>s into a state (I mean `items`). You can learn the case in the screenshot :)